### PR TITLE
Implement customized mosip auth design

### PR DESCRIPTION
**Pull request by ldclakmal**
## Purpose
This is the implementation of customized MOSIP auth design as in https://github.com/mosip/websub/blob/develop/src/filters/authfilter.bal

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: websubhub-examples/kafka-hub/hub/hub_service.bal [hunk 1]
```diff
@@ -40,7 +40,7 @@ service object {
     isolated remote function onRegisterTopic(websubhub:TopicRegistration message, http:Headers headers)
                                 returns websubhub:TopicRegistrationSuccess|websubhub:TopicRegistrationError|error {
         if config:SECURITY_ON {
-            check security:authorize(headers, ["register_topic"]);
+            check security:authorizeSubscriber(headers, message.topic);
```
Comment (ayeshLK) [inline websubhub-examples/kafka-hub/hub/hub_service.bal:43]: Should not this be `security:authorizePublisher(headers, message.topic)` ?

File: websubhub-examples/kafka-hub/hub/hub_service.bal [hunk 2]
```diff
@@ -68,7 +68,7 @@ service object {
     isolated remote function onDeregisterTopic(websubhub:TopicDeregistration message, http:Headers headers)
                         returns websubhub:TopicDeregistrationSuccess|websubhub:TopicDeregistrationError|error {
         if config:SECURITY_ON {
-            check security:authorize(headers, ["deregister_topic"]);
+            check security:authorizeSubscriber(headers, message.topic);
```
Comment (ayeshLK) [inline websubhub-examples/kafka-hub/hub/hub_service.bal:71]: ditto

Comment (ldclakmal) [reply websubhub-examples/kafka-hub/hub/hub_service.bal]: Fixed with https://github.com/ballerina-platform/module-ballerina-websubhub/pull/236/commits/c11d3faf75b7f8021b70197983a1dc38b64a36b4

File: websubhub-examples/kafka-hub/hub/hub_service.bal (hunk 1 above)
Comment (ldclakmal) [reply websubhub-examples/kafka-hub/hub/hub_service.bal]: Fixed with https://github.com/ballerina-platform/module-ballerina-websubhub/pull/236/commits/c11d3faf75b7f8021b70197983a1dc38b64a36b4

**Review by ayeshLK — APPROVED** [review]

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**